links/src/utils/generateLinks.ts

```diff
@@ -10,8 +11,25 @@ type GenerateLinks = {
  */
 export const generateLinks: GenerateLinks = async () => {
 	const links = await getLinks();
+	const arr = Object.values(links);
+	let tree = totree(arr, 0);
 
-	// TODO: logic
+	fs.writeFileSync("./output.json", JSON.stringify(tree, null, 2));
 
 	return [];
 };
+
+export const totree = (arr: any[], parent_id) => {
+	let tree = [];
+	for (const node of arr) {
+		// use key value to get parent
+		if (node.parent_id === parent_id) {
+			var children = totree(arr, node.id);
+			if (children.length) {
+				node.children = children;
+			}
+			tree.push(node);
+		}
+	}
+	return tree;
+};
```

Comment:
```suggestion
export const totree = (arr: any[], parent_id: number = 0) => {
	let tree = [];
	for (const node of arr) {
		// use key value to get parent
		if (node.parent_id === parent_id) {
			let children = totree(arr, node.id);
			if (children.length) {
				node.children = children;
			}
			tree.push(node);
		}
	}
	return tree;
};
```
One can define a default value for parent_id in the parameters. Then you don't have to use it for the root parent

`var` is an older JavaScript variable assignment. Common standards are `let` and `const`